### ISSUE TITLE
[🐛] Not correct detection Python poetry project

### ISSUE THREAD

**Issue by ansidorov** [snyk/snyk#1802]
- `node -v`: `v15.14.0`
- `npm -v`: `7.7.6`
- `snyk -v`: `1.530.0`
- OS: `Ubuntu 20.04`
- Command run: `snyk test --command=python3 --all-projects`

It is not quite correct to assume by the presence of `pyproject.toml` file that Poetry is used in the project, because it is something like a new setup.cfg, which can store configurations of many different tools, it is standardised directly by the python developers in a number of PEPs:
https://www.python.org/dev/peps/pep-0517/
https://www.python.org/dev/peps/pep-0518/
https://www.python.org/dev/peps/pep-0621/

You should look in the `pyproject.toml` file for the presence of a block for Poetry to indicate that the project uses it.

### Expected behaviour
Detect only one project from Pipfile.lock

### Actual behaviour
Snyk detected 2 projects: Pipfile and poetry.

### Steps to reproduce
```
# git clone https://github.com/ansidorov/snyk-poetry-bug.git
# cd snyk-poetry-bug
# docker build -t snyk .
# docker run --rm -ti -v $(pwd):/app -w /app snyk:latest bash
# pipenv install
# snyk auth
# snyk test --command=python3 --all-projects
✗ 1/2 potential projects failed to get dependencies. Run with `-d` for debug output.

Testing /app...

Organization:      my_org
Package manager:   pip
Target file:       Pipfile
Project name:      app
Open source:       no
Project path:      /app
Licenses:          enabled

✓ Tested 6 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.
```

### Debug log
If applicable, please append the `--debug` flag on your command and include the output here **ensuring to remove any sensitive/personal details or tokens**. 

```
# snyk test --command=python3 --all-projects -d
  snyk test <ref *1> {
  _: [ [Circular *1] ],
  command: 'python3',
  debug: true,
  allProjects: true
} +0ms
  snyk-test auto detect manifest files, found 2 [ '/app/Pipfile', '/app/pyproject.toml' ] +0ms
  snyk-test all dependencies for app project dir
  snyk-test ✗ Failed to get dependencies for /app/pyproject.toml
  snyk-test ERROR: Cannot find lockfile /app/poetry.lock
  snyk-test  +0ms
  snyk-test Found 1 projects from 2 detected manifests +2s
✗ 1/2 potential projects failed to get dependencies. Run with `-d` for debug output.
  snyk Potential policy locations found: [ '/app' ] +0ms
  snyk:run-test converting dep-tree to dep-graph { name: 'app', targetFile: 'Pipfile' } +0ms
  snyk:run-test done converting dep-tree to dep-graph { uniquePkgsCount: 7 } +8ms
  snyk:prune rootPkg { name: 'app', version: '0.0.0' } +0ms
  snyk:prune prePrunePathsCount: 7 +0ms
  snyk:prune isDenseGraph false +0ms
  snyk sending request to: https://snyk.io/api/v1/test-dep-graph +0ms
  snyk request body size: 1339 +0ms
  snyk gzipped request body size: 416 +0ms
  snyk not using proxy +4ms

Testing /app...

Organization:      myorg
Package manager:   pip
Target file:       Pipfile
Project name:      app
Open source:       no
Project path:      /app
Licenses:          enabled

✓ Tested 6 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.
  snyk analytics {
  "args": [
    {
      "command": "python3",
      "debug": true,
      "allProjects": true
    }
  ],
  "command": "test",
  "metadata": {
    "local": true,
    "allProjects": {
      "scannedProjects": 1,
      "targetFiles": [
        "/app/Pipfile",
        "/app/pyproject.toml"
      ],
      "packageManagers": [
        "pip",
        "poetry"
      ],
      "ignore": [
        "node_modules"
      ]
    },
    "pluginName": "custom-auto-detect",
    "policies": 1,
    "policyLocations": [
      "/app"
    ],
    "packageManager": "pip",
    "packageName": "app",
    "packageVersion": "0.0.0",
    "package": "app@0.0.0",
    "prePrunedPathsCount": 7,
    "payloadSize": 1339,
    "gzippedPayloadSize": 416,
    "depGraph": true,
    "isDocker": false,
    "vulns-pre-policy": 0,
    "vulns": 0
  },
  "version": "1.530.0",
  "os": "Linux 4.19",
  "nodeVersion": "v15.14.0",
  "standalone": false,
  "integrationName": "",
  "integrationVersion": "",
  "integrationEnvironment": "",
  "integrationEnvironmentVersion": "",
  "id": "3f6b955bb902651a3fff6ec4fdfbfdd4b1efa0e7",
  "ci": false,
  "environment": {
    "npmVersion": "7.7.6"
  },
  "durationMs": 3186,
  "metrics": {
    "network_time": {
      "type": "timer",
      "values": [
        580
      ],
      "total": 580
    },
    "cpu_time": {
      "type": "synthetic",
      "values": [
        2606
      ],
      "total": 2606
    }
  }
} +0ms
  snyk sending request to: https://snyk.io/api/v1/analytics/cli +1s
  snyk request body size: 998 +0ms
  snyk gzipped request body size: 531 +0ms
  snyk not using proxy +1ms
```

**Comment by jan-stehlik** [comment]
hey @ansidorov, thanks for an excellent bug description, we are looking into this now and will let you know once it's fixed

**Comment by jan-stehlik** [comment]
hey @ansidorov the fix is now released in version [v1.585.0](https://github.com/snyk/snyk/releases/tag/v1.585.0)
